prime-router/src/main/kotlin/credentials/HashicorpVaultCredentialService.kt

```diff
@@ -1,44 +1,76 @@
 package gov.cdc.prime.router.credentials
 
-import com.github.kittinunf.fuel.core.FuelManager
+import gov.cdc.prime.router.common.HttpClientUtils
+import io.ktor.client.HttpClient
+import io.ktor.http.HttpStatusCode
 import org.apache.logging.log4j.kotlin.Logging
 import org.json.JSONObject
 
 internal object HashicorpVaultCredentialService : CredentialService(), Logging {
 
     private val VAULT_API_ADDR: String by lazy { System.getenv("VAULT_API_ADDR") ?: "http://127.0.0.1:8200" }
     private val VAULT_TOKEN: String by lazy { System.getenv("VAULT_TOKEN") ?: "" }
-    private val manager: FuelManager by lazy { initVaultApi(FuelManager()) }
 
-    internal fun initVaultApi(manager: FuelManager): FuelManager {
-        // Set the default path
-        manager.basePath = "$VAULT_API_ADDR/v1/secret"
+    override fun fetchCredential(
+        connectionId: String,
+    ): Credential? {
+        return fetchCredentialHelper(
+            connectionId = connectionId
+        )
+    }
 
-        // Initialize the API token used to secure requests
-        manager.baseHeaders = mapOf("X-Vault-Token" to VAULT_TOKEN)
+    /**
+     * object specific impl - also adapted to unit tests
+     */
+    fun fetchCredentialHelper(
+        connectionId: String,
+        httpClient: HttpClient? = null,
```

Comment:
I don't understand this change. httpClient is only used once below and it is ignore and overwritten?